stonesoup/plotter.py

```diff
@@ -921,3 +926,369 @@ def plot_sensors(self, sensors, sensor_label="Sensors", **kwargs):
+            self.plotting_data.append(_AnimationPlotterDataClass(
+                plotting_data=[State(state_vector=plotting_state_vector,
+                                     timestamp=detection.timestamp)
+                               for detection, plotting_state_vector in plot_clutter.items()],
+                plotting_label=measurements_label + " clutter",
+                plotting_keyword_arguments=clutter_kwargs
+            ))
+
+    def plot_sensors(self, sensors, sensor_label="Sensors", **kwargs):
+        raise NotImplementedError
+
+    @classmethod
+    def run_animation(cls,
+                      times_to_plot: List[datetime],
+                      data: Iterable[_AnimationPlotterDataClass],
+                      plot_item_expiry: Optional[timedelta] = None,
+                      axis_padding: float = 0.1,
+                      figure_kwargs: dict = {},
+                      animation_input_kwargs: dict = {},
+                      x_label: str = "$x$",
+                      y_label: str = "$y$"
+                      ) -> animation.FuncAnimation:
+        """
+        Parameters
+        ----------
+        times_to_plot : Iterable[datetime]
+            All the times, that the plotter should plot
+        data : Iterable[datetime]
+            All the data that should be plotted
+        plot_item_expiry: timedelta
+            How long a state should be displayed for. None means the
+        axis_padding: float
+            How much extra space should be given around the edge of the plot
+        figure_kwargs: dict
+            Keyword arguments for the pyplot figure function. See matplotlib.pyplot.figure for more
+            details
+        animation_input_kwargs: dict
+            Keyword arguments for FuncAnimation class. See matplotlib.animation.FuncAnimation for
+            more details. Default values are: blit=False, repeat=False, interval=50
+        x_label: str
+            Label for the x axis
+        y_label: str
+            Label for the y axis
+
+        Returns
+        -------
+        : animation.FuncAnimation
+            Animation object
+        """
+
+        animation_kwargs = dict(blit=False, repeat=False, interval=50)  # milliseconds
+        animation_kwargs.update(animation_input_kwargs)
+
+        fig1 = plt.figure(**figure_kwargs)
+
+        the_lines = []
+        plotting_data = []
+        legends_key = []
+
+        for a_plot_object in data:
+            if a_plot_object.plotting_data is not None:
+                the_data = np.array(
+                    [a_state.state_vector for a_state in a_plot_object.plotting_data])
+                if len(the_data) == 0:
+                    continue
+                the_lines.append(
+                    plt.plot([],  # the_data[:1, 0],
+                             [],  # the_data[:1, 1],
+                             **a_plot_object.plotting_keyword_arguments)[0])
+
+                legends_key.append(a_plot_object.plotting_label)
```

Comment:
Wonder if you could filter this legend list so you only have one item for each unique label?

Reply:
Currently looks like this: ![image](https://user-images.githubusercontent.com/19266036/197992097-88fff50d-c9d6-4be6-a2a5-73a42b2a68d5.png)

Reply:
The legend should be less cluttered now. 

I didn't want to use unique labels in the `run_animation` function in case the user wanted the same label for two different line types. Instead when using `plot_tracks` or `plot_ground_truths` only the first track/truth is given a label. Multiple calls of the either plotting function could result in multiple identical labels however I think thismatches the behaviour of the standard `Plotter`

Reply:
Looks good! One last comment (sorry) - could you set the legend position using `plt.legend(loc='upper left'` or similar? Would stop it jumping around in the animation.

Reply:
Good point. The random clutter distribution meant that it was jumping around a lot. 

I've added the option for the user to specify keyword arguments for the legend function. In the example I've specified the the legend location should top left.